compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -175,7 +175,7 @@ object desugar {
       case Into(tpt1) =>
         mods1 = vdef.mods.withAddedAnnotation(
           TypedSplice(
-            Annotation(defn.AllowConversionsAnnot).tree.withSpan(tpt.span.startPos)))
+            Annotation(defn.AllowConversionsAnnot, tpt.span.startPos).tree.withSpan(tpt.span.startPos)))
```

Comment:
I would imagine that the second `withSpan` is not needed anymore.

```suggestion
            Annotation(defn.AllowConversionsAnnot, tpt.span.startPos).tree))
```